data/campaigns/campaignManager.ts

```diff
@@ -100,7 +100,21 @@ class CampaignManager implements ICampaignManager {
     this.logger.debug('shouldShowCampaign', {centerId, campaignId, location, atDate});
     const campaign = CAMPAIGNS[campaignId];
     const date = atDate ?? new Date();
-    const campaignActive = campaign.enabled && date >= campaign.startDate && date < campaign.endDate;
+
+    // For typescript, the type of `locations` is inferred as a union of *all* the locations
```

Comment:
Changing line 101 to

```ts
const campaign: CampaignLocations<T> = CAMPAIGNS[campaignId];
```

Leads to the error:

```
The intersection

     {
     readonly enabled: true;
     readonly allowedCenters: readonly ["NWAC"];
     readonly locations: {
         readonly 'observation-list-view': {
             readonly frequency: 0;
             readonly startDate: Date;
             readonly endDate: Date;
         };
         readonly 'map-view': { ...; };     }; } & { ...; } & { ...; } & { ...; } & { ...; }

was reduced to never because property enabled has conflicting types in some constituents.
```

IMHO using the inferred type here and getting stuff like `readonly enabled: true;` is weird. Maybe just using the explicit type declarations will help?